numpy/lib/tests/test_function_base.py

```diff
@@ -2773,8 +2773,8 @@ def test_period(self):
 
 
 def compare_results(res, desired):
-    for i in range(len(desired)):
-        assert_array_equal(res[i], desired[i])
+    for result_item, desired_item in zip(res, desired):
+        assert_array_equal(result_item, desired_item)
```

Comment:
Does this need a `if len(res) != len(desired): raise ...` or is it guarenteed that they will be equal length in all the tests?

Reply:
It seems this function is not used, so it is now removed.